src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs

```diff
@@ -198,14 +200,22 @@ public void DialogSendKeys(string dialogAutomationName, string keys)
             SendKeys(new KeyPress(VirtualKey.K, ShiftState.Ctrl), new KeyPress(VirtualKey.F, ShiftState.Ctrl));
         }
 
-        public void Paste(string text)
+        private void ExecuteOnSTAThread(ThreadStart a)
         {
-            var thread = new Thread(() => Clipboard.SetText(text));
+            var thread = new Thread(a);
             thread.SetApartmentState(ApartmentState.STA);
             thread.Start();
             thread.Join();
+        }
 
+        public void Paste(string text)
+        {
+            ExecuteOnSTAThread(() => Clipboard.SetText(text));
             VisualStudioInstance.Dte.ExecuteCommand("Edit.Paste");
+
+            // Formatting paste tests sometimes fail with nothing pasted.
+            // Verify we're actually pasted the righ text.
+            ExecuteOnSTAThread(() => Assert.Equal(text, Clipboard.GetText()));
```

Comment:
Can we do Set Text, Verify Clipboard Contents (repeat if that API is flaky), _then_ actually do the Edit.Paste?